File: es1/es1.go
package main

import (
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	freq := 0
	var drift int
	for {
		_, err := fmt.Fscanf(input, "%d\n", &drift)
		if err != nil {
			break
		}
		freq += drift
	}
	fmt.Println("Final frequency", freq)
}

File: es1/es1.c
#include <ctype.h>
#include <stdio.h>

int main(int argc, char *argv[])
{
	if (argc < 2) {
		fprintf(stderr, "Usage: %s filename\n", argv[0]);
		return -1;
	}

	FILE *input = fopen(argv[1], "rb");
	if (input == NULL) {
		fprintf(stderr, "File %s not found\n", argv[1]);
		return -1;
	}

	int freq = 0;
	int drift = 0;
	while (1) {
		if (fscanf(input, " %d", &drift) == EOF)
			break;
		freq += drift;
		drift = 0;
	}
	fclose(input);

	fprintf(stdout, "Final frequency: %d\n", freq);
	return 0;
}
